quidel_covidtest/delphi_quidel_covidtest/run.py

```diff
@@ -55,6 +57,45 @@ def get_smooth_info(sensors, _SMOOTHERS):
             smoothers[sensor] = smoothers.pop(RAW_TEST_PER_DEVICE)
     return smoothers
 
+def generate_and_export_for_nonparent_geo(geo_groups, res_key, smooth, device,
+                                          first_date, last_date, suffix, # generate args
+                                          geo_res, sensor_name, export_dir,
+                                          export_start_date, export_end_date, # export args
+                                          log_filename, log_exceptions): # logger args
+    """Generate sensors, create export CSV then return stats."""
+    logger = get_structured_logger(__name__, log_filename, log_exceptions)
```

Comment:
Because trying to pass the existing logger breaks multiprocessing:
```
AttributeError: Can't pickle local object 'BoundLoggerLazyProxy.bind.<locals>.finalized_bind'"
```

Reply:
damn, i should have seen that coming...  :(
this is worth a comment in the code, something like ```# `logger` cannot be passed to a child process```

however!  this current implementation with the newly constructed loggers is susceptible to concurrency issues, as they are all going to be trying to write to the same file from different processes without a mutex.  to alleviate this, you should be able to pass along a [`multiprocessing.Lock()`](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Lock) to use as a context manager around logging operations.  that does add some more verbosity to your `generate_and_...()` calls, but it will work for now -- i have an idea for something that is cleaner and reusable for situations like this in the future.

Reply:
i just created #1885 which could greatly simplify the logging stuff, removing the need for the lock and re-creating loggers, as soon as thats available